src/main/scala/org/broadinstitute/dsde/agora/server/webservice/QueryHandler.scala

```diff
@@ -59,8 +68,14 @@ class QueryHandler(authorizationProvider: AuthorizationProvider) extends Actor {
             entityTypes: Seq[AgoraEntityType.EntityType],
             username: String): Unit = {
     val entities = agoraBusiness.find(agoraSearch, agoraProjection, entityTypes, username)
-    authorizationProvider.filterByReadPermissions(authorizationProvider.authorizationsForEntities(entities, username))
-    context.parent ! RequestComplete(entities)
+    try {
+      val authorizedEntities = authorizationProvider.filterByReadPermissions(authorizationProvider.authorizationsForEntities(entities, username))
+      context.parent ! RequestComplete(authorizedEntities)
+    }
+    catch {
+      case clientServiceFailure: ClientServiceFailure => context.parent ! RequestComplete(InternalServerError, "Error encountered in entity authorization")
+      case timeoutException: TimeoutException => context.parent! RequestComplete(InternalServerError, "Timeout exception encountered during entity authorization")
+    }
```

Comment:
Jay and I talked in person about various ways to refactor this. None were attractive. Leaving as-is. We can circle back and clean it up when implementing write or edit if people have any ideas